storage/beacondb/eth1.go

```diff
@@ -22,38 +22,32 @@ package beacondb
 
 import (
 	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
-	"github.com/berachain/beacon-kit/primitives/bytes"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/version"
 )
 
 // GetLatestExecutionPayloadHeader retrieves the latest execution payload
 // header from the BeaconStore.
 func (kv *KVStore) GetLatestExecutionPayloadHeader() (
 	*ctypes.ExecutionPayloadHeader, error,
 ) {
-	// NOTE: unmarshalling this struct is NOT affected by it's own fork version. The versioned
-	// codec is left in for backwards compatibility.
-	forkVersion, err := kv.latestExecutionPayloadVersion.Get(kv.ctx)
-	if err != nil {
-		return nil, err
-	}
-	kv.latestExecutionPayloadCodec.SetActiveForkVersion(bytes.FromUint32(forkVersion))
 	return kv.latestExecutionPayloadHeader.Get(kv.ctx)
 }
 
 // SetLatestExecutionPayloadHeader sets the latest execution payload header in
 // the BeaconStore.
 func (kv *KVStore) SetLatestExecutionPayloadHeader(
-	payloadHeader *ctypes.ExecutionPayloadHeader,
+	payloadHeader *ctypes.ExecutionPayloadHeader, forkVersion common.Version,
 ) error {
-	// NOTE: marshalling this struct is NOT affected by it's own fork version. The versioned
-	// codec is left in for backwards compatibility.
-	version := payloadHeader.GetForkVersion()
-	if err := kv.latestExecutionPayloadVersion.Set(
-		kv.ctx, version.ToUint32(),
-	); err != nil {
-		return err
+	// NOTE: marshalling this struct is NOT affected by it's own fork version.
+	if version.IsBefore(forkVersion, version.Electra()) {
```

Comment:
Can delay the hard fork by making this `version.Electra1()` or something else in the future

Reply:
This refactored code is still backwards compatible

Reply:
super nit: consider inverting test to simplfiy code. Something like

```golang
if !version.IsBefore(forkVersion, version.Electra()) {
  // post Electra we simplified marshalling and we do not need to store version
  return nil
}
...
```